packages/node_modules/@webex/internal-plugin-device/test/unit/spec/device.js

```diff
@@ -178,5 +179,164 @@ describe('plugin-device', () => {
         );
       });
     });
+
+    describe('#refresh()', () => {
+      let requestSpy;
+
+      const setup = () => {
+        sinon.stub(device, 'canRegister').callsFake(() => Promise.resolve());
+        sinon.stub(device, 'processRegistrationSuccess').callsFake(() => {});
+        requestSpy = sinon.spy(device, 'request');
+        device.config.defaults = {};
+        device.set('registered', true);
+      };
+
+      it('If-None-Match header is added if etag is set', async () => {
+        setup();
+
+        device.set('etag', 'etag-value');
+
+        const result = device.refresh();
+
+        await result;
+
+        assert.deepEqual(requestSpy.args[0][0].headers, {
+          'If-None-Match': 'etag-value'
+        });
+      });
+
+      it('If-None-Match header is not added if etag is not set', async () => {
+        setup();
+
+        const result = device.refresh();
+
+        await result;
+
+        assert.deepEqual(requestSpy.args[0][0].headers, {});
+      });
+    });
+
+    describe('#processRegistrationSuccess()', () => {
+      let clonedDTO;
+
+      const setup = () => {
+        clonedDTO = cloneDeep(dto);
+        clonedDTO.features = {
+          developer: [
+            {
+              key: '1',
+              type: 'boolean',
+              val: 'true',
+              value: true,
+              mutable: true,
+              lastModified: '2015-06-29T20:02:48.033Z'
+            },
+          ],
+          entitlement: [
+            {
+              key: '2',
+              val: 'true',
+              value: true,
+              mutable: false
+            }
+          ],
+          user: [
+            {
+              key: '3',
+              val: 'true',
+              value: true,
+              mutable: true
+            }
+          ]
+        };
+      };
+
+      const checkFeatureNotPresent = (type, key) => {
+        assert.deepEqual(device.features[type].get(key), undefined);
+      };
+
+      const checkFeature = (type, key, expectedValue) => {
+        assert.equal(device.features[type].length, 1);
+        assert.deepEqual(device.features[type].get(key).get('value'), expectedValue);
+      };
+
+      it('features are set correctly if etag not in headers', () => {
+        setup();
+        const response = {
+          body: {
+            ...clonedDTO
+          },
+          headers: {
+
+          }
+        };
+
+        checkFeatureNotPresent('developer', '1');
+        checkFeatureNotPresent('entitlement', '2');
+        checkFeatureNotPresent('user', '3');
+
+        device.processRegistrationSuccess(response);
+
+        checkFeature('developer', '1', true);
+        checkFeature('entitlement', '2', true);
+        checkFeature('user', '3', true);
+      });
```

Comment:
suggestion: Probably worth checking the etag remains `undefined` at the end of this (though it clearly couldn't pick up a value from anywhere else...)